Sources/StreamChat/Utils/SystemEnvironment+UserAgent.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+#if os(iOS)
+import UIKit
+#elseif os(macOS)
+import IOKit
+#endif
+
+extension SystemEnvironment {
+    private static var info: [String: Any] { Bundle.main.infoDictionary ?? [:] }
+    private static var appName: String {
+        ((info["CFBundleDisplayName"] ?? info[kCFBundleIdentifierKey as String]) as? String) ?? "App name unavailable"
+    }
+
+    private static var appVersion: String { (info["CFBundleShortVersionString"] as? String) ?? "0" }
```

Comment:
This is just personal style, but I think these one liners are not easily readable

Reply:
Updated